### PR TITLE
Adapt sensor idlharness tests to latest spec

### PR DESCRIPTION
**Pull request by Honry**
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/4793)

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision 376082a72befd08f10de68d0cd70df2c2a722e8d
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="https://w3c-test.org/submissions/4793/accelerometer/idlharness.https.html">/accelerometer/idlharness.https.html</a></summary>


|                                                     Subtest                                                      | Results |
|------------------------------------------------------------------------------------------------------------------|---------|
|                                                                                                                  | OK      |
| `Accelerometer interface: existence and properties of interface object`                                          | FAIL    |
| `Accelerometer interface object name`                                                                            | FAIL    |
| `Accelerometer interface: attribute includesGravity`                                                             | FAIL    |
| `Accelerometer interface object length`                                                                          | FAIL    |
| `Stringification of new Accelerometer();`                                                                        | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "y" with the proper type (1)`               | FAIL    |
| `Accelerometer interface: existence and properties of interface prototype object`                                | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "includesGravity" with the proper type (3)` | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "state" with the proper type (0)`                  | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "onerror" with the proper type (6)`                | FAIL    |
| `Accelerometer interface: existence and properties of interface prototype object's "constructor" property`       | FAIL    |
| `Accelerometer must be primary interface of new Accelerometer();`                                                | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "timestamp" with the proper type (1)`              | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "start" with the proper type (2)`                  | FAIL    |
| `Accelerometer interface: attribute x`                                                                           | FAIL    |
| `Accelerometer interface: attribute y`                                                                           | FAIL    |
| `Accelerometer interface: attribute z`                                                                           | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "onchange" with the proper type (4)`               | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "x" with the proper type (0)`               | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "z" with the proper type (2)`               | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "stop" with the proper type (3)`                   | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "onactivate" with the proper type (5)`             | FAIL    |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/4793/gyroscope/idlharness.https.html">/gyroscope/idlharness.https.html</a></summary>


|                                                Subtest                                                 | Results |
|--------------------------------------------------------------------------------------------------------|---------|
|                                                                                                        | OK      |
| `Stringification of new Gyroscope();`                                                                  | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "onactivate" with the proper type (5)`       | FAIL    |
| `Gyroscope interface: existence and properties of interface prototype object`                          | FAIL    |
| `Gyroscope interface: attribute y`                                                                     | FAIL    |
| `Gyroscope interface: attribute z`                                                                     | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "timestamp" with the proper type (1)`        | FAIL    |
| `Gyroscope interface object length`                                                                    | FAIL    |
| `Gyroscope interface: new Gyroscope(); must inherit property "x" with the proper type (0)`             | FAIL    |
| `Gyroscope interface: existence and properties of interface prototype object's "constructor" property` | FAIL    |
| `Gyroscope interface: existence and properties of interface object`                                    | FAIL    |
| `Gyroscope interface: new Gyroscope(); must inherit property "z" with the proper type (2)`             | FAIL    |
| `Gyroscope interface: new Gyroscope(); must inherit property "y" with the proper type (1)`             | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "stop" with the proper type (3)`             | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "onerror" with the proper type (6)`          | FAIL    |
| `Gyroscope interface object name`                                                                      | FAIL    |
| `Gyroscope interface: attribute x`                                                                     | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "onchange" with the proper type (4)`         | FAIL    |
| `Gyroscope must be primary interface of new Gyroscope();`                                              | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "start" with the proper type (2)`            | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "state" with the proper type (0)`            | FAIL    |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/4793/ambient-light/idlharness.https.html">/ambient-light/idlharness.https.html</a></summary>


|                                                        Subtest                                                         | Results |
|------------------------------------------------------------------------------------------------------------------------|---------|
|                                                                                                                        | OK      |
| `Stringification of new AmbientLightSensor();`                                                                         | FAIL    |
| `AmbientLightSensor interface object length`                                                                           | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "onactivate" with the proper type (5)`              | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "onchange" with the proper type (4)`                | FAIL    |
| `AmbientLightSensor interface: existence and properties of interface object`                                           | FAIL    |
| `AmbientLightSensor must be primary interface of new AmbientLightSensor();`                                            | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "stop" with the proper type (3)`                    | FAIL    |
| `AmbientLightSensor interface: existence and properties of interface prototype object's "constructor" property`        | FAIL    |
| `AmbientLightSensor interface object name`                                                                             | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "start" with the proper type (2)`                   | FAIL    |
| `AmbientLightSensor interface: new AmbientLightSensor(); must inherit property "illuminance" with the proper type (0)` | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "state" with the proper type (0)`                   | FAIL    |
| `AmbientLightSensor interface: attribute illuminance`                                                                  | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "timestamp" with the proper type (1)`               | FAIL    |
| `AmbientLightSensor interface: existence and properties of interface prototype object`                                 | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "onerror" with the proper type (6)`                 | FAIL    |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/4793/magnetometer/idlharness.https.html">/magnetometer/idlharness.https.html</a></summary>


|                                                  Subtest                                                  | Results |
|-----------------------------------------------------------------------------------------------------------|---------|
|                                                                                                           | OK      |
| `Magnetometer interface: existence and properties of interface object`                                    | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "onerror" with the proper type (6)`          | FAIL    |
| `Magnetometer interface: new Magnetometer(); must inherit property "y" with the proper type (1)`          | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "onchange" with the proper type (4)`         | FAIL    |
| `Magnetometer interface: existence and properties of interface prototype object`                          | FAIL    |
| `Magnetometer interface: existence and properties of interface prototype object's "constructor" property` | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "timestamp" with the proper type (1)`        | FAIL    |
| `Magnetometer interface: attribute z`                                                                     | FAIL    |
| `Magnetometer interface object length`                                                                    | FAIL    |
| `Magnetometer interface: attribute x`                                                                     | FAIL    |
| `Magnetometer interface: attribute y`                                                                     | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "stop" with the proper type (3)`             | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "onactivate" with the proper type (5)`       | FAIL    |
| `Magnetometer interface: new Magnetometer(); must inherit property "x" with the proper type (0)`          | FAIL    |
| `Magnetometer interface object name`                                                                      | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "start" with the proper type (2)`            | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "state" with the proper type (0)`            | FAIL    |
| `Stringification of new Magnetometer();`                                                                  | FAIL    |
| `Magnetometer interface: new Magnetometer(); must inherit property "z" with the proper type (2)`          | FAIL    |
| `Magnetometer must be primary interface of new Magnetometer();`                                           | FAIL    |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision 376082a72befd08f10de68d0cd70df2c2a722e8d
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="https://w3c-test.org/submissions/4793/accelerometer/idlharness.https.html">/accelerometer/idlharness.https.html</a></summary>


|                                                     Subtest                                                      | Results |
|------------------------------------------------------------------------------------------------------------------|---------|
|                                                                                                                  | OK      |
| `Accelerometer interface: existence and properties of interface object`                                          | FAIL    |
| `Accelerometer interface object name`                                                                            | FAIL    |
| `Accelerometer interface: attribute includesGravity`                                                             | FAIL    |
| `Accelerometer interface object length`                                                                          | FAIL    |
| `Stringification of new Accelerometer();`                                                                        | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "y" with the proper type (1)`               | FAIL    |
| `Accelerometer interface: existence and properties of interface prototype object`                                | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "includesGravity" with the proper type (3)` | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "state" with the proper type (0)`                  | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "onerror" with the proper type (6)`                | FAIL    |
| `Accelerometer interface: existence and properties of interface prototype object's "constructor" property`       | FAIL    |
| `Accelerometer must be primary interface of new Accelerometer();`                                                | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "timestamp" with the proper type (1)`              | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "start" with the proper type (2)`                  | FAIL    |
| `Accelerometer interface: attribute x`                                                                           | FAIL    |
| `Accelerometer interface: attribute y`                                                                           | FAIL    |
| `Accelerometer interface: attribute z`                                                                           | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "onchange" with the proper type (4)`               | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "x" with the proper type (0)`               | FAIL    |
| `Accelerometer interface: new Accelerometer(); must inherit property "z" with the proper type (2)`               | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "stop" with the proper type (3)`                   | FAIL    |
| `Sensor interface: new Accelerometer(); must inherit property "onactivate" with the proper type (5)`             | FAIL    |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/4793/gyroscope/idlharness.https.html">/gyroscope/idlharness.https.html</a></summary>


|                                                Subtest                                                 | Results |
|--------------------------------------------------------------------------------------------------------|---------|
|                                                                                                        | OK      |
| `Stringification of new Gyroscope();`                                                                  | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "onactivate" with the proper type (5)`       | FAIL    |
| `Gyroscope interface: existence and properties of interface prototype object`                          | FAIL    |
| `Gyroscope interface: attribute y`                                                                     | FAIL    |
| `Gyroscope interface: attribute z`                                                                     | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "timestamp" with the proper type (1)`        | FAIL    |
| `Gyroscope interface object length`                                                                    | FAIL    |
| `Gyroscope interface: new Gyroscope(); must inherit property "x" with the proper type (0)`             | FAIL    |
| `Gyroscope interface: existence and properties of interface prototype object's "constructor" property` | FAIL    |
| `Gyroscope interface: existence and properties of interface object`                                    | FAIL    |
| `Gyroscope interface: new Gyroscope(); must inherit property "z" with the proper type (2)`             | FAIL    |
| `Gyroscope interface: new Gyroscope(); must inherit property "y" with the proper type (1)`             | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "stop" with the proper type (3)`             | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "onerror" with the proper type (6)`          | FAIL    |
| `Gyroscope interface object name`                                                                      | FAIL    |
| `Gyroscope interface: attribute x`                                                                     | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "onchange" with the proper type (4)`         | FAIL    |
| `Gyroscope must be primary interface of new Gyroscope();`                                              | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "start" with the proper type (2)`            | FAIL    |
| `Sensor interface: new Gyroscope(); must inherit property "state" with the proper type (0)`            | FAIL    |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/4793/ambient-light/idlharness.https.html">/ambient-light/idlharness.https.html</a></summary>


|                                                        Subtest                                                         | Results |
|------------------------------------------------------------------------------------------------------------------------|---------|
|                                                                                                                        | OK      |
| `Stringification of new AmbientLightSensor();`                                                                         | FAIL    |
| `AmbientLightSensor interface object length`                                                                           | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "onactivate" with the proper type (5)`              | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "onchange" with the proper type (4)`                | FAIL    |
| `AmbientLightSensor interface: existence and properties of interface object`                                           | FAIL    |
| `AmbientLightSensor must be primary interface of new AmbientLightSensor();`                                            | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "stop" with the proper type (3)`                    | FAIL    |
| `AmbientLightSensor interface: existence and properties of interface prototype object's "constructor" property`        | FAIL    |
| `AmbientLightSensor interface object name`                                                                             | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "start" with the proper type (2)`                   | FAIL    |
| `AmbientLightSensor interface: new AmbientLightSensor(); must inherit property "illuminance" with the proper type (0)` | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "state" with the proper type (0)`                   | FAIL    |
| `AmbientLightSensor interface: attribute illuminance`                                                                  | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "timestamp" with the proper type (1)`               | FAIL    |
| `AmbientLightSensor interface: existence and properties of interface prototype object`                                 | FAIL    |
| `Sensor interface: new AmbientLightSensor(); must inherit property "onerror" with the proper type (6)`                 | FAIL    |

</details>
<details>
<summary><a href="https://w3c-test.org/submissions/4793/magnetometer/idlharness.https.html">/magnetometer/idlharness.https.html</a></summary>


|                                                  Subtest                                                  | Results |
|-----------------------------------------------------------------------------------------------------------|---------|
|                                                                                                           | OK      |
| `Magnetometer interface: existence and properties of interface object`                                    | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "onerror" with the proper type (6)`          | FAIL    |
| `Magnetometer interface: new Magnetometer(); must inherit property "y" with the proper type (1)`          | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "onchange" with the proper type (4)`         | FAIL    |
| `Magnetometer interface: existence and properties of interface prototype object`                          | FAIL    |
| `Magnetometer interface: existence and properties of interface prototype object's "constructor" property` | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "timestamp" with the proper type (1)`        | FAIL    |
| `Magnetometer interface: attribute z`                                                                     | FAIL    |
| `Magnetometer interface object length`                                                                    | FAIL    |
| `Magnetometer interface: attribute x`                                                                     | FAIL    |
| `Magnetometer interface: attribute y`                                                                     | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "stop" with the proper type (3)`             | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "onactivate" with the proper type (5)`       | FAIL    |
| `Magnetometer interface: new Magnetometer(); must inherit property "x" with the proper type (0)`          | FAIL    |
| `Magnetometer interface object name`                                                                      | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "start" with the proper type (2)`            | FAIL    |
| `Sensor interface: new Magnetometer(); must inherit property "state" with the proper type (0)`            | FAIL    |
| `Stringification of new Magnetometer();`                                                                  | FAIL    |
| `Magnetometer interface: new Magnetometer(); must inherit property "z" with the proper type (2)`          | FAIL    |
| `Magnetometer must be primary interface of new Magnetometer();`                                           | FAIL    |

</details>

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/4793/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/225919594)

# Lint #

Passed

**Comment by Honry** [comment]
@tobie, could you help review this PR? All the Web IDL tests for Sensors are update-to-date.

File: orientation-sensor/idlharness.https.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+};
+
+[Constructor(optional SensorOptions sensorOptions)]
+interface AbsoluteOrientationSensor : OrientationSensor {
+};
+</pre>
+
+<script>
+
+(() => {
```
Comment (tobie) [inline orientation-sensor/idlharness.https.html]: Are there guidelines wrt what flavor of ES should be used?

Comment (Honry) [inline orientation-sensor/idlharness.https.html]: No, AFAIK.

Comment (tobie) [inline orientation-sensor/idlharness.https.html]: OK, so I checked on irc ans the answer is: _"Has to work in latest Chrome, Fx, Safari, Edge"_.

So you're good.

**Review by tobie — APPROVED** [review]
So seems mostly like a copy/paste job from the spec, so LGTM. :)

**Pull request merged**